main/src/test/java/org/mobilitydata/gtfsvalidator/validator/OverlappingFrequencyValidatorTest.java

```diff
@@ -0,0 +1,114 @@
+                .isEmpty();
+    }
+
+    @Test
+    public void validSequentialReversed() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "07:00:00", "10:00:00", 300),
+                createFrequency(3, "t0", "05:00:00", "07:00:00", 600)))
+                .isEmpty();
+    }
+
+    @Test
+    public void validWithGap() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "08:00:00", "10:00:00", 300)))
+                .isEmpty();
+    }
+
+    @Test
+    public void validDifferentTrips() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t1", "06:00:00", "10:00:00", 300)))
+                .isEmpty();
+    }
+
+    @Test
+    public void overlappingPartially() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "06:00:00", "10:00:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        2, GtfsTime.fromString("07:00:00"),
+                        3, GtfsTime.fromString("06:00:00"), "t0"));
+    }
+
+    @Test
+    public void overlappingIncludedSameStart() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "05:00:00", "06:30:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        3, GtfsTime.fromString("06:30:00"),
+                        2, GtfsTime.fromString("05:00:00"), "t0"));
+    }
+
+    @Test
+    public void overlappingIncludedSameEnd() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "07:00:00", 600),
+                createFrequency(3, "t0", "06:30:00", "07:00:00", 300)))
+                .containsExactly(new OverlappingFrequencyNotice(
+                        2, GtfsTime.fromString("07:00:00"),
+                        3, GtfsTime.fromString("06:30:00"), "t0"));
+    }
+
+    @Test
+    public void overlappingThreeIntervals() {
+        assertThat(validateFrequencies(
+                createFrequency(2, "t0", "05:00:00", "05:25:00", 600),
+                createFrequency(3, "t0", "05:00:00", "05:15:00", 300),
+                createFrequency(4, "t0", "05:20:00", "05:40:00", 300)))
+                .containsExactly(
+                        new OverlappingFrequencyNotice(
+                                3, GtfsTime.fromString("05:15:00"),
+                                2, GtfsTime.fromString("05:00:00"), "t0"),
+                        new OverlappingFrequencyNotice(
+                                2, GtfsTime.fromString("05:25:00"),
+                                4, GtfsTime.fromString("05:20:00"), "t0"));
+    }
+}
```

Comment:
Since we're testing stop times here, do you think we should include at a least one test for when times are greater than 24:00:00? Seems like a corner case (at least by conventional time calculations) that may be worth covering.

Reply:
I do not think that it is worth testing. That is not a corner case for this validator. If we really want to check that 24:00:00 is properly compared to 01:00:00, then we should place this to `GtfsTimeTest.java`.

Reply:
As someone who's stumbled over the slightly unusual semantics of GTFS `Time` before and seen many, many wrong implementations, I think it would be cool to very that this project handles them properly. During DST switches, [it is especially unintuitive](https://gist.github.com/derhuerst/574edc94981a21ef0ce90713f1cff7f6).

Reply:
DST is not related to frequency validation. The GTFS time abstracts out all these details.

Yes, representing time is a hard task but this validation is much simpler and it does not require tests for:

* DST
* Gregorian vs Julian calendar
* Leap seconds
* Time after 24:00
* Timezones
* 12 clock format vs 24 clock
* AM vs PM
* Year 2000 problem
* End of Maya calendar
* Leap years

Reply:
Ah, so you're purely validating based on the "nominal" value, not on it's "effects" on the referenced trips?

After thinking about it, that makes sense, because one `frequency.txt` entry references one trip, which usually has *many* service days.